### PR TITLE
[aws] Ensure event.kind is correctly set for pipeline errors

### PR DESCRIPTION
**Pull request by MakoWish**
## What does this PR do?

Modify aws integration to correctly set `event.kind` for pipeline errors and ensure `error.message` is an array.

## Checklist

- [X] I have reviewed tips for building integrations and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's changelog.yml file.
- [x] I have verified that Kibana version constraints are current according to guidelines.
 
## Author's Checklist

- [x]  

## Related issues

* For #6582

## Screenshots

### CODE REVIEW

File: packages/aws/data_stream/cloudtrail/elasticsearch/ingest_pipeline/third-party.yml [hunk 1]
```diff
@@ -23,6 +23,9 @@ processors:
       field: event.original
       target_field: json
 on_failure:
+  - set:
+      field: event.kind
+      value: pipeline_error
   - append:
       field: error.message
       value: >-
```
Comment (bhapas) [inline packages/aws/data_stream/cloudtrail/elasticsearch/ingest_pipeline/third-party.yml:31]: Should the value not change here as well?

Comment (MakoWish) [reply packages/aws/data_stream/cloudtrail/elasticsearch/ingest_pipeline/third-party.yml:31]: What value are you expecting?

Comment (bhapas) [reply packages/aws/data_stream/cloudtrail/elasticsearch/ingest_pipeline/third-party.yml:31]:
```diff
- value: >-
-   error in third party api pipeline:
-   error in [{{_ingest.on_failure_processor_type}}] processor{{#_ingest.on_failure_processor_tag}}
-   with tag [{{_ingest.on_failure_processor_tag }}]{{/_ingest.on_failure_processor_tag}}
-   {{ _ingest.on_failure_message }}
+ value: '{{{ _ingest.on_failure_message }}}'

```

Comment (MakoWish) [reply packages/aws/data_stream/cloudtrail/elasticsearch/ingest_pipeline/third-party.yml:31]: I don't think there needs to be a standard error message. It's more that any pipeline errors populate `event.kind: pipeline_error`, and that there is an `error.message` of some sort that would detail the pipeline error.

File: packages/aws/data_stream/rds/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -278,3 +278,81 @@ processors:
       field: aws.rds.metrics.AuroraVolumeBytesLeftTotal.avg
       target_field: aws.rds.aurora_volume_left_total.bytes
       ignore_missing: true
+  - remove:
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/rds/elasticsearch/ingest_pipeline/default.yml:281]: @zmoog Correct me if Im wrong please, I thought rename automatically remove the old fields. So no need to add `remove` here anymore.

Comment (MakoWish) [reply packages/aws/data_stream/rds/elasticsearch/ingest_pipeline/default.yml:281]: I am not sure why that `remove` processor is showing as added in my branch. I did not add those. Did another PR remove them, so now my changes appear to be adding them again?

Comment (kaiyan-sheng) [reply packages/aws/data_stream/rds/elasticsearch/ingest_pipeline/default.yml:281]: ahh ok lets remove them from this PR then :) Thanks!

Comment (MakoWish) [reply packages/aws/data_stream/rds/elasticsearch/ingest_pipeline/default.yml]: Done! :-)

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request closed without merging**